pkg/urbit/noun/events.c

```diff
@@ -404,29 +404,30 @@ _ce_patch_delete(void)
 static c3_o
 _ce_patch_verify(u3_ce_patch* pat_u)
 {
-  c3_w i_w, pag_w, mug_w;
-  c3_w      mem_w[pag_wiz_i];
-  size_t    off_i, siz_i = pag_siz_i;
-  ssize_t   ret_i;
+  c3_w  pag_w, mug_w;
+  c3_w  mem_w[pag_wiz_i];     /* ;;: This is a VLA, you know... */
```

Comment:
That's a very good point. It didn't used to be, but was switched (unintentionally I think) in #5881.

Reply:
And obviously the same goes for any other arrays declared with `pag_wiz_i` or `pag_siz_i`. Instances in `_ce_patch_apply`, `_ce_image_fine`, `_ce_image_copy`, etc. In this case, yeah a macro actually is better. Or we could adopt c23 and start using the new `constexpr` type. ;)

Reply:
A `const` variable and a `#define` generate identical assembly when compiled with `-O2` or greater, so it's not `mem_w` isn't a variable-length array in release builds. That said, you're right that a macro is probably preferable.

Here's a simple example demonstrating that `const` and `#define` generate identical assembly at `-O2` and above:

```console
$ gcc --version
gcc (GCC) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
$ cat << EOF > main.c
#if defined(CONST)
static const unsigned int len = 4;
#elif defined(MACRO)
# define len 4
#endif

int
main(int argc, char *argv[])
{
  char arr[len];
  arr[0] = 'a';
  arr[1] = 'b';
  arr[2] = 'c';
  arr[3] = 'd';
  return 0;
}
EOF
$ gcc -DCONST -O2 main.c -o const && objdump -d const > const.objdump
$ gcc -DMACRO -O2 main.c -o macro && objdump -d macro > macro.objdump
$ diff const.objdump macro.objdump
2c2
< const:     file format elf64-x86-64
---
> macro:     file format elf64-x86-64
```

Reply:
Yep, confirming that these arrays only show up as a VLA on unoptimized (or -O1 -- didn't test) builds. It's my understanding that gcc/clang tends to make VLA initialization an `alloca` call when it can. But this doesn't even matter much in these use cases as they're functionally identical.

Being a VLA would be a bigger deal if these were globals, because we'd be unnecessarily wasting stack space instead of storing it in bss.

Reply:
All that code is synchronous -- there probably should just be a single, global "page-sized scratch buffer".